### PR TITLE
Add support for NAS addresses

### LINKED ISSUE #270: Add support for Nebulas (NAS) addresses in encoding library
Here is the encoding library: https://github.com/ensdomains/address-encoder

Example: https://github.com/ensdomains/address-encoder/pull/47/files

Discord for dev help: https://discord.com/invite/AskZbFx

Please provide a reference link so we can check you did the encoding correctly.

We'd like to keep the encoding library as small as possible, so please be efficient in your use of space.

We will likely be reviewing PRs and paying out bounties more or less on a weekly basis. No need to ping us to check your PR, we will definitely get to it.

Thanks for your help!

### PR DESCRIPTION
**Pull request by Echolon166**
## Issue number

Closes #270 

## Description

Add support for NAS addresses

## Reference to the specification

- https://github.com/nebulasio/wiki/blob/master/address.md
- https://github.com/nebulasio/neb.js/blob/e0f25d71e10a384e15f77b0cc2edd4ffb1c41fbc/lib/account.js#L85

## Reference to the test address.

- https://github.com/nebulasio/neb.js/blob/e0f25d71e10a384e15f77b0cc2edd4ffb1c41fbc/test/account.test.js#L20

## List of features added/changed

- Implement NAS encoder
- Implement NAS decoder

## Checklist:

- [x] My code follows the code style of this project.
- [x] My code implements all the required features.
- [x] I have specified correct coinTypes specified at [Slip 44](https://github.com/satoshilabs/slips/blob/master/slip-0044.md)
- [x] I have provided the reference link to the specification I implemented.
- [x] I have provided enough explanation about how I provided the test address

### CODE REVIEW

File: src/index.ts [hunk 1]
```diff
@@ -29,8 +29,9 @@ import {
 } from 'crypto-addr-codec';
 import { sha512_256 } from 'js-sha512';
 import { decode as nanoBase32Decode, encode as nanoBase32Encode } from 'nano-base32';
+import { check } from 'prettier';
```
Comment (makoto) [inline src/index.ts:32]: Where is this in use?

File: src/__tests__/index.test.ts [hunk 2]
```diff
@@ -838,6 +838,13 @@ const vectors: Array<TestVector> = [
       { text: 'EQDZ4T6YyVkg9mb2cAuLEu8iBKbajQAywF', hex: '214d797cc92303dac242b17026e79bbea28eb642f29f0d3582' }
     ]
   },
+  {
+    name: 'NAS',
+    coinType: 2718,
+    passingVectors: [
+      { text: 'n1FF1nz6tarkDVwWQkMnnwFPuPKUaQTdptE', hex: '07f964ff495324635f22c7b486e05d7e67c7af5c' },
```
Comment (makoto) [inline src/__tests__/index.test.ts:845]: Can you also add `n1sLnoc7j57YfzAVP8tJ3yK5a2i56QrTDdK` to test that the account address also works?

File: src/index.ts (hunk 1 above)
Comment (Echolon166) [reply src/index.ts]: Seems like I mistakenly added it, fixed.

File: src/__tests__/index.test.ts (hunk 2 above)
Comment (Echolon166) [reply src/__tests__/index.test.ts]: Added the test case.

Was adding/removing first 2 bytes while encoding/decoding the address which was [unnecessary](https://github.com/nebulasio/go-nebulas/blob/be27d30e2eab5d1b7dab13e21456f52981acecbf/core/address_test.go#L42), so fixed that.

**Pull request merged**